staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -146,3 +150,45 @@ type CustomResourceValidation struct {
 	// OpenAPIV3Schema is the OpenAPI v3 schema to be validated against.
 	OpenAPIV3Schema *JSONSchemaProps
 }
+
+// CustomResourceSubResources defines the status and scale subresources for CustomResources.
+type CustomResourceSubResources struct {
+	// Status denotes the status subresource for CustomResources
+	Status *CustomResourceSubResourceStatus
+	// Scale denotes the scale subresource for CustomResources
+	Scale *CustomResourceSubResourceScale
+}
+
+// CustomResourceSubResourceStatus defines how to serve the status subresource for CustomResources.
+// Status is represented by the `.status` JSON path inside of a CustomResource. When set,
+// * exposes a /status subresource for the custom resource
+// * PUT requests to the /status subresource take a custom resource object, and ignore changes to anything except the status stanza
+// * PUT/POST/PATCH requests to the custom resource ignore changes to the status stanza
+type CustomResourceSubResourceStatus struct{}
+
+// CustomResourceSubResourceScale defines how to serve the scale subresource for CustomResources.
+type CustomResourceSubResourceScale struct {
+	// SpecReplicasPath defines the JSON path inside of a CustomResource that corresponds to Scale.Spec.Replicas.
+	// Only JSON paths without the array notation are allowed.
+	// Must be a JSON Path under .spec.
+	// If there is no value under the given path in the CustomResource, the /scale subresource will return an error on GET.
+	SpecReplicasPath string
+	// StatusReplicasPath defines the JSON path inside of a CustomResource that corresponds to Scale.Status.Replicas.
+	// Only JSON paths without the array notation are allowed.
+	// Must be a JSON Path under .status.
+	// If there is no value under the given path in the CustomResource, the status replica value in the /scale subresource
+	// will default to 0.
+	StatusReplicasPath string
+	// LabelSelectorPath defines the JSON path inside of a CustomResource that corresponds to Scale.Status.Selector.
+	// Only JSON paths without the array notation are allowed.
+	// Must be a JSON Path under .status.
+	// If there is no value under the given path in the CustomResource, the status label selector value in the /scale
+	// subresource will default to the empty string.
+	// +optional
+	LabelSelectorPath string
+	// ScaleGroupVersion denotes the GroupVersion in the form "group/version"
+	// of the Scale object sent as the payload for /scale.
+	// It allows transition to future versions easily.
+	// Today only autoscaling/v1 is allowed.
+	ScaleGroupVersion string
```

Comment:
I don't understand how this helps future version transitions. Can you expand the comment?

Reply:
We have to fix the `Scale` GroupVersion per CRD version. Today there is just `autoscaling/v1`. Some day we have `autoscaling/v2`. Then it must be controllable which `Scale` type is used. I.e. it's less about switching, but more about fixing it per version. When we have conversion and multiple versions in the future, the scale group version can be different for each CRD version.

Something like "It allows to support other payload types in the future. Today only autoscaling/v1 is allowed." is better than the current sentence.